scripts-local/tests/integration_tests.bats

```diff
@@ -141,78 +155,124 @@ setup() {
+  diff=$(($NEW_STAKE - $OLD_STAKE))
+  assert_equal "$diff" "333"
 }
 
 # check that a second liquid staking call kicks off reinvestment
-@test "[NOT-IMPLEMENTED] rewards are being reinvested" {
-  # check the rewards balance
-  # wait a day
-  # check the withdrawal account balance
-  # wait a day
-  # check that 90% of rewards were transferred to the delegation account
-  # check that 10% of rewards were transferred to the rev EOA
-  # wait a day
-  # check that rewards were staked
+@test "[INTEGRATION-BASIC] rewards are being reinvested (delegated balance increasing)" {
+  old_staked_bal=$($GAIA_CMD q staking delegation $DELEGATION_ICA_ADDR $GAIA_DELEGATE_VAL | GETSTAKE)
+
+  # liquid stake again to kickstart the reinvestment process
+  $STRIDE_CMD tx stakeibc liquid-stake 1000 uatom --keyring-backend test --from val1 -y --chain-id $STRIDE_CHAIN
+  # wait four days (transfers, stake, move rewards, reinvest rewards)
+  remaining_seconds=$($STRIDE_CMD q epochs seconds-remaining day)
+  sleep $remaining_seconds
+  BLOCK_SLEEP 2  
+  remaining_seconds=$($STRIDE_CMD q epochs seconds-remaining day)
+  sleep $remaining_seconds
+  remaining_seconds=$($STRIDE_CMD q epochs seconds-remaining day)
+  sleep $remaining_seconds
+  remaining_seconds=$($STRIDE_CMD q epochs seconds-remaining day)
+  sleep $remaining_seconds
+
+  # simple check that number of tokens staked increases
+  new_staked_bal=$($GAIA_CMD q staking delegation $DELEGATION_ICA_ADDR $GAIA_DELEGATE_VAL | GETSTAKE)
+  STAKED_BAL_INCREASED=$((new_staked_bal > old_staked_bal))
+  assert_equal "$STAKED_BAL_INCREASED" "1"
+}
+
+# check that exchange rate is updating
+@test "[INTEGRATION-BASIC] exchange rate is updating" {
+  # read the exchange rate
+  RR1=$($STRIDE_CMD q stakeibc list-host-zone | grep -Fiw 'RedemptionRate' | grep -Eo '[+-]?[0-9]+([.][0-9]+)?')
+
+  # wait for reinvestment to happen (3 days is enough)
+  remaining_seconds=$($STRIDE_CMD q epochs seconds-remaining day)
+  sleep $remaining_seconds
+  remaining_seconds=$($STRIDE_CMD q epochs seconds-remaining day)
+  sleep $remaining_seconds
+  remaining_seconds=$($STRIDE_CMD q epochs seconds-remaining day)
+  sleep $remaining_seconds
+
+  RR2=$($STRIDE_CMD q stakeibc list-host-zone | grep -Fiw 'RedemptionRate' | grep -Eo '[+-]?[0-9]+([.][0-9]+)?')
+
+  # check that the exchange rate has increased
+  MULT=1000000
+  RR_INCREASED=$(( $(FLOOR $(DECMUL $RR2 $MULT)) > $(FLOOR $(DECMUL $RR1 10000))))
+  assert_equal "$RR_INCREASED" "1"
 }
 
+
 # check that redemptions and claims work
-@test "[NOT-IMPLEMENTED] redemption works" {
+@test "[INTEGRATION-BASIC] redemption works" {
+  old_redemption_ica_bal=$($GAIA_CMD q bank balances $REDEMPTION_ICA_ADDR --denom uatom | GETBAL)
   # call redeem-stake
-  # check for an unbonding record
-  # check that a UserRedemptionRecord was created with isClaimabled = false
+  amt_to_redeem=89
+  $STRIDE_CMD tx stakeibc redeem-stake $amt_to_redeem GAIA cosmos1g6qdx6kdhpf000afvvpte7hp0vnpzapuyxp8uf \
```

Comment:
is this the ICQ relayer address? If so, why is that be used instead of one of the validator addresses?

Reply:
this is a random address (might be the icq relayer, not sure) chosen at random simply to check the functionality of redeem-stake